platforms/core-execution/build-cache/src/main/java/org/gradle/caching/internal/controller/DefaultBuildCacheController.java

```diff
@@ -140,7 +141,7 @@ private Optional<BuildCacheLoadResult> loadRemoteAndStoreResultLocally(BuildCach
             return Optional.empty();
         }
         AtomicReference<Optional<BuildCacheLoadResult>> result = new AtomicReference<>(Optional.empty());
-        tmp.withTempFile(key, file -> {
+        tmp.withTempFile(((DefaultBuildCacheKey) key).getHashCodeInternal(), file -> {
```

Comment:
🤔 Do we need hashcode in tmp.withTempFile, or could we just pass use string here?

Reply:
We could, but I think it's best to keep the richer type here, as it's more descriptive of the purpose of this data.